apps/system/js/call_forwarding.js

```diff
@@ -85,7 +85,16 @@
       var index = slot.index;
       var simCard = slot.simCard;
 
-      if (this._callForwardingIconInitializedStates[index] || !simCard) {
+      if (this._callForwardingIconInitializedStates[index]) {
+        return;
+      }
+
+      var that = this;
+      if (!simCard) {
+        that._callForwardingHelper.get(function(states) {
+          states[index] = false;
```

Comment:
For the cases where iccid is not available, we should also hide the icon when the card state is not `ready` and icc id is unavailable.